plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt

```diff
@@ -166,20 +172,33 @@ open class KtlintPlugin : Plugin<Project> {
 
     private fun createConfiguration(target: Project, extension: KtlintExtension) =
             target.configurations.maybeCreate("ktlint").apply {
-                target.dependencies.add(this.name,
-                        mapOf("group" to "com.github.shyiko", "name" to "ktlint", "version" to extension.version))
+                target.dependencies.add(
+                    this.name,
+                    mapOf(
+                        "group" to "com.github.shyiko",
+                        "name" to "ktlint",
+                        "version" to extension.version
+                    )
+                )
             }
 
     private fun addAdditionalRunArgs(extension: KtlintExtension, runArgs: MutableSet<String>) {
         if (extension.verbose) runArgs.add("--verbose")
         if (extension.debug) runArgs.add("--debug")
+        addAndroidArg(extension, runArgs)
+        if (extension.ruleSets.size > 0) {
+            extension.ruleSets.forEach { runArgs.add("--ruleset=$it") }
+        }
+    }
+
+    private fun addAndroidArg(
+        extension: KtlintExtension,
+        runArgs: MutableSet<String>
+    ) {
         if (extension.android && SemVer.parse(extension.version) >= SemVer(0, 12, 0)) {
             // Android option is available from ktlint 0.12.0
             runArgs.add("--android")
         }
-        if (extension.ruleSets.size > 0) {
-            extension.ruleSets.forEach { runArgs.add("--ruleset=$it") }
-        }
     }
```

Comment:
General question, why is a mutable collection passed through all of this code instead of having each method append all of these list together as each method returns?

It seems weird to have a mutable list floating through these various methods.

Not related to your code changes directly, I just realized that this was kind of a weird practice going on .